### DISCUSSION TITLE
Programmatically scroll the chart on a swipe gesture

### DISCUSSION THREAD

**Discussion by Praveen-Android** [patrykandpatrick/vico#930] [Questions]
### Question

**Sample Usecase**
- I have a graph that has 10 ticks from x = 1 to x = 10.
- I have a zoomstate set to 3 so that only 3 ticks/labels are shown at a time
```
rememberVicoZoomState(
        zoomEnabled = false,
        initialZoom =  3.0,
        minZoom =  3.0,
        maxZoom =  3.0
)
```
- I have the scroll state disabled but have provided a way to scroll back / forward (by 3 data points from the example) using buttons and with programmatic scroll/animate scroll (`scrollState.animateScroll / scrollState.scroll`)

**Question**
I want to listen in to swipe gesture to perform programmatic scroll (by 3 here) similar to what i am doing with buttons. Searched the documentation but couldn't find any references or may be i missed if there was any. How do i go about intercepting swipe gestures in a graph.

### Vico version(s)

2.0.0-alpha.28

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @Praveen-Android. Just to clarify, do you mean detecting horizontal drag gestures, and then scrolling the chart on pointer up? Or, scrolling the chart together with the pointer, and then snapping?
